### PR TITLE
feature flags added to allow a very minimal install

### PR DESCRIPTION
**Pull request by paulczar**
* feature flags to allow disabling of logging/monitoring
* ensure feature flags for cinder etc work as expected

* example of using those flags in envs/example/jeofd to install
  a minimal openstack footprint.

(jeofd = just enough openstack for docker)

### CODE REVIEW

File: roles/cinder-control/tasks/main.yml [hunk 1]
```diff
@@ -35,3 +35,4 @@
   ufw: rule=allow to_port=8778 proto=tcp
 
 - include: monitoring.yml tags=monitoring,common
+  when: monitoring.enabled is not defined or monitoring.enabled|bool
```
Comment (j2sol) [inline roles/cinder-control/tasks/main.yml]: To clean this up a bit, why not `monitoring.enabled|default('yes')|bool` ?

File: roles/heat/tasks/main.yml [hunk 2]
```diff
@@ -54,8 +54,10 @@
     - heat-engine
 
 - include: monitoring.yml tags=monitoring,common
+  when: monitoring.enabled is not defined or monitoring.enabled|bool
 
 - include: logging.yml
   tags:
     - logrotate
     - logging
+  when: logging.enabled is not defined or logging.enabled|bool
```
Comment (j2sol) [inline roles/heat/tasks/main.yml]: Same for these ones, `when: logging.enabled|default('true')|bool`

**Comment by j2sol** [comment]
Looks like the envs/example/defaults.yml does not have a logging: top level hash, so that would need to be created.

File: roles/nova-data/tasks/main.yml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ---
 - include: ssh.yml
-  when: nova.enable_ssh|bool
+  when: nova.enable_ssh|when: monitoring.enabled|default(True)|bool
```
Comment (j2sol) [inline roles/nova-data/tasks/main.yml]: :confused:  Not sure what's going on here, pretty sure it's not planned.

**Comment by paulczar** [comment]
retest

**Pull request merged**